api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -181,23 +184,23 @@ else if (!isObsGroup) {
 				} else if (dt.isNumeric()) {
 					ConceptNumeric cn = Context.getConceptService().getConceptNumeric(c.getConceptId());
 					// If the concept numeric is not precise, the value cannot be a float, so raise an error 
-					if (!cn.getAllowDecimal() && Math.ceil(obs.getValueNumeric()) != obs.getValueNumeric()) {
+					if (cn != null && !cn.getAllowDecimal() && Math.ceil(obs.getValueNumeric()) != obs.getValueNumeric()) {
```

Comment:
Did you get cases where the data type was numeric but cn was null?

Reply:
Yes, I found out this with some failing tests (existing ones).

Reply:
I would really love to see the failures (such that we unmask that bug and fix it) because we do not expect null concept numerics whenever the data type is numeric.

Reply:
I will remove the null checks and hopefully the errors can bubble up again.